### DISCUSSION TITLE
Need help with state management

### DISCUSSION THREAD

**Discussion by mueen-memon** [xyflow/xyflow#2224] [Q&A] [answered]
Hi,
I am trying to use zustand state management but it is in type script. I have my codebase in JS. 
Thanks.

**Comment by moklick** [comment]
Hey @mueen-memon 

you don't need typescript for using Zustand. What is the issue?

**Reply by mueen-memon** [reply]
I needed the js version of the code sandbox.

**Comment by kahboom** [comment, marked as answer]
@mueen-memon - @moklick is right, you don't need TS for Zustand. I guess some of the confusion comes from the typings provided on the website. It's nearly exactly the same except you remove the typings.

For the store:

```
import create from 'zustand';
import {
  Connection,
  Edge,
  EdgeChange,
  Node,
  NodeChange,
  addEdge,
  OnNodesChange,
  OnEdgesChange,
  OnConnect,
  applyNodeChanges,
  applyEdgeChanges,
} from 'react-flow-renderer';

import initialNodes from './nodes';
import initialEdges from './edges';

const useStore = create((set, get) => ({
  nodes: initialNodes,
  edges: initialEdges,
  onNodesChange: (changes) => {
    set({
      nodes: applyNodeChanges(changes, get().nodes),
    });
  },
  onEdgesChange: (changes) => {
    set({
      edges: applyEdgeChanges(changes, get().edges),
    });
  },
  onConnect: (connection) => {
    set({
      edges: addEdge(connection, get().edges),
    });
  },
}));

export default useStore;
```

And then for `nodes.ts` and `edges.ts` just remove the ` as Edge[]` casting at the end of each file;

**Reply by mueen-memon** [reply]
Thanks.
